### ISSUE TITLE
Bug: lock stopped working with "Device internal error due to device states not synchronized with server"

### ISSUE THREAD

**Issue by Jacksonbm1** [OpenWonderLabs/homebridge-switchbot#435]
### Describe The Bug

The lock won't unlock via homekit/homebridge with error :190

### To Reproduce

Unlock door using homekit tile (generated via homebridge)

### Expected behavior

Door should unlock

### Relevant log output

```shell
[7/30/2022, 1:08:23 AM] [SwitchBot] Lock: Front door lock Sending request to SwitchBot API. command: unlock, parameter: default, commandType: command
[7/30/2022, 1:08:23 AM] [SwitchBot] Lock: Front door lock Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"not support device type"} format is invalid

This has never happened before and the lock was working fine yesterday.
```

### Config for homebridge-switchbot

```shell
"name": "SwitchBot",
            "credentials": {
                "openToken": "<redacted>",
                "notice": "Keep your token a secret!"
            },
            "platform": "SwitchBot
```

### Device & Model

It's just the lock

### Node.js Version

v16.16.0

### NPM Version

v8.14.0

### Homebridge Version

v1.5.0

### Homebridge Switchbot Plugin Version

v1.14.1

### Homebridge Config UI X Plugin Version

v4.50.0

### Operating System

Raspbian

**Comment by Jacksonbm1** [comment]
Add details: The lock works with the switchbot app with/without wifi/bluetooth, which means that it's not the cloud integration's fault.

**Comment by Jacksonbm1** [comment]
Rebooted the hub as per https://github.com/OpenWonderLabs/SwitchBotAPI/issues/3#issuecomment-751431339 but it did not resolve the issue.

**Comment by Jacksonbm1** [comment]
Found related issue: https://github.com/OpenWonderLabs/SwitchBotAPI/issues/121

**Comment by guptaramm** [comment]
Can we connect with the dev team and ask the status ?

**Comment by Jacksonbm1** [comment]
https://twitter.com/richardmou63/ is one of the devs and is very responsive to twitter/reddit dms. Also their facebook group. I would also follow the related issue that I commented in: https://github.com/OpenWonderLabs/homebridge-switchbot/issues/435#issuecomment-1200094783

**Comment by drosenberg-gh** [comment]
I purchased a SwitchBot lock because of the API and specifically this plug-in. I was really disappointed to hear that SwitchBot disabled the ability to control the lock via the API (OpenWonderLabs/SwitchBotAPI#121). At the very least can you please enable the ability to lock the SwitchBot lock using the API/Homebridge plug-in.

**Comment by tgibs97** [comment]
Are there any updates on this?

**Comment by Jacksonbm1** [comment]
This isn't really an issue with the plugin. It's an issue with the api that the plugin uses. This is the link to that bug: https://github.com/OpenWonderLabs/SwitchBotAPI/issues/121

You should follow it there. 

Edited: corrected link

**Comment by tgibs97** [comment]
Ohhhh I see now, yeah looks like Switchbot turned off API access for their lock, that sucks, thanks for the clarification

**Comment by Jacksonbm1** [comment]
Hello, this issue should be reopened now that the integration has been fixed: https://github.com/OpenWonderLabs/SwitchBotAPI/issues/121

**Comment by donavanbecker** [comment]
if you can help https://github.com/OpenWonderLabs/homebridge-switchbot/pull/474

**Comment by jamesalmeida** [comment]
I hope this can be fixed now!

**Comment by donavanbecker** [comment]
@jamesalmeida If I get time today. It should be done soon

**Comment by donavanbecker** [comment]
@jamesalmeida are you able to help at all? stuck with implementing the new API Auth for v1.1

**Comment by jamesalmeida** [comment]
> @jamesalmeida are you able to help at all? stuck with implementing the new API Auth for v1.1

I would if I could, but I'm not experienced enough to implement it. I'm trying to learn more about how these plugins work though.

**Comment by donavanbecker** [comment]
@jamesalmeida getting an Unauthorized Request: [here](https://github.com/OpenWonderLabs/homebridge-switchbot/blob/36985c34074c34af45b944d27fea5f799c4714cb/src/platform.ts#L79-L86)

If you can help with that at all.

**Comment by donavanbecker** [comment]
See https://github.com/OpenWonderLabs/homebridge-switchbot/issues/476

**Comment by donavanbecker** [comment]
Fixed with v2.0.0, Once released.
